src/Instrumentation/ReactHttp/src/ReactHttpInstrumentation.php

```
@@ -0,0 +1,159 @@
+
+                /**
+                 * This cannot be tested with PHPUnit without modifying php.ini with:
+                 * otel.instrumentation.http.request_headers[]="Accept"
+                 */
+                // @codeCoverageIgnoreStart
+                foreach ((array) (get_cfg_var('otel.instrumentation.http.request_headers') ?: []) as $header) {
+                    if ($request->hasHeader($header)) {
+                        $spanBuilder->setAttribute(
+                            sprintf('%s.%s', TraceAttributes::HTTP_REQUEST_HEADER, strtolower($header)),
+                            $request->getHeaderLine($header)
+                        );
+                    }
+                }
+                // @codeCoverageIgnoreEnd
+
+                $span = $spanBuilder->startSpan();
+                $context = $span->storeInContext($parentContext);
+                $propagator->inject($request, HeadersPropagator::instance(), $context);
+
+                Context::storage()->attach($context);
+
+                return [$request];
+            },
+            /** @psalm-suppress UnusedClosureParam */
+            post: static function (Transaction $transaction, array $params, PromiseInterface $promise, ?Throwable $exception): PromiseInterface {
+                $scope = Context::storage()->scope();
+                $scope?->detach();
+
+                if (!$scope || $scope->context() === Context::getCurrent()) {
+                    return $promise;
+                }
+
+                $span = Span::fromContext($scope->context());
+
+                return $promise->then(
+                    onFulfilled: function (ResponseInterface $response) use ($span) {
+                        $span
+                            ->setAttribute(TraceAttributes::HTTP_RESPONSE_STATUS_CODE, $response->getStatusCode())
+                            ->setAttribute(TraceAttributes::HTTP_RESPONSE_BODY_SIZE, $response->getHeaderLine('Content-Length'));
```

This header isn't always there, is that an issue or will it just silently accept it: https://github.com/reactphp/http/blob/fc5f0a6739ba4a10de3847bab4688c602e5fe948/src/Io/Sender.php#L82-L94

---

It will silently accept it, an empty string is allowed. However, I think we should not set it if it's not known.

---

I removed this in the upcoming commit; the equivalent data can be opted-in by the library user with the php.ini setting: `otel.instrumentation.http.response_headers[]="Content-Length"`